ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java

```diff
@@ -36,48 +36,56 @@
  * This class represents the API server. On start-up it deploys multiple server verticles to spread
  * the load across available cores.
  */
+// CHECKSTYLE_RULES.OFF: ClassDataAbstractionCoupling
 public class Server {
+  // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
 
   private static final Logger log = LoggerFactory.getLogger(Server.class);
 
   private final Vertx vertx;
-  private final JsonObject config;
+  private final ApiServerConfig config;
   private final Endpoints endpoints;
-  private final HttpServerOptions httpServerOptions;
   private final Map<ApiQueryID, QuerySubscriber> queries = new ConcurrentHashMap<>();
   private final Set<HttpConnection> connections = new ConcurrentHashSet<>();
   private String deploymentID;
 
-  public Server(final Vertx vertx, final JsonObject config, final Endpoints endpoints,
-      final HttpServerOptions httpServerOptions) {
+  public Server(final Vertx vertx, final ApiServerConfig config, final Endpoints endpoints) {
     this.vertx = Objects.requireNonNull(vertx);
     this.config = Objects.requireNonNull(config);
     this.endpoints = Objects.requireNonNull(endpoints);
-    this.httpServerOptions = Objects.requireNonNull(httpServerOptions);
+  }
+
+  private HttpServerOptions createHttpServerOptions(final ApiServerConfig apiServerConfig) {
+    return
+        new HttpServerOptions().setHost(apiServerConfig.getString(ApiServerConfig.LISTEN_HOST))
+            .setPort(apiServerConfig.getInt(ApiServerConfig.LISTEN_PORT))
+            .setUseAlpn(true)
+            .setSsl(true)
```

Comment:
I would of thought we'd want ssl to be optional... could drive it off the presence or absence of `ApiServerConfig.KEY_PATH`?

Reply:
Yes, we could. But HTTP2 is usually always used with TLS - all major browsers require TLS with HTTP2. I know we're not aiming at the browser case but perhaps we should follow suit?